metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -153,6 +153,14 @@ def partition_id_text(self) -> Optional[str]:
         return None
 
 
+@dataclasses.dataclass
+class BrowsePath:
+    file: str
+    timestamp: datetime
+    size: int
+    partitions: List[Folder]
+
```

Comment:
Nice! So browser would browse through file system to yield one BrowsePath for one identified table and that table would then be emitted as dataset if its allowed by patterns, right ?

Reply:
yes, exactly